### DISCUSSION TITLE
addEventListener() does not work well on Safari

### DISCUSSION THREAD

**Discussion by mlbonniec** [vercel/next.js#13929] [Help]
# Bug report

## Describe the bug
In development mode, the `console.log()` (*see just below*) is done on Firefox, but not on Safari. In production mode, the `console.log()` is displayed on both browsers.

## To Reproduce
Put an event listener in the `componentDidMount()` of your React class component.
```js
componentDidMount() {
    window.addEventListener("load", () => console.log("load"));
}
```

## Expected behavior
I'd like to get my event fired up every time.

## System information
- OS: macOS Catalina 10.15.1
- Browser Safari 13.0.3 and Firefox Developer Edition 78.0b3
- Version of Next.js: 9.4.1
- Version of Node.js: 12.16.1

**Comment by mahish** [comment]
If you check `document.readyState` in Safari, you realize it returns `complete`. So the page was loaded even before it added the load event listener. To make it work use this:
```javascript
if (document.readyState === 'complete') {
  callback();
} else {
  window.addEventListener('load', callback);
}
```
